alpa/torch/__init__.py

```diff
@@ -0,0 +1,140 @@
+"""Miscellaneous functions available in `alpa.torch.*` namespace
+"""
+try:
+    import torch
+except ImportError as e:
+    print("""
+Attempted to use Alpa-PyTorch frontend, but PyTorch is not installed.
+
+Please follow instructions at docs/torch_build_instructions.txt
+to install PyTorch and related dependencies.
+""")
+    raise e
+
+from typing import Any, Callable, Union, Tuple
+from functools import partial, wraps
+
+import numpy as np
+import alpa
+from alpa.device_mesh import DistributedArray
+from alpa.torch.nn import functionalize, materialize, meta_init
+from alpa.torch.ops.mapping import enable_dist_for_funcs, zeros_like_on_device
+from alpa.torch.tensor_utils import make_shaped_array_from_pt_tensor, \
+    meta_like, to_format, assert_format
+
+# If True, prints verbose log for debugging.
+debug = False
+
+
+def set_mode(new_mode: str):
+    """This sets the current Alpa mode. Supports one of following:
+
+    "local":
+    - Pure PT eager mode on single CPU/GPU
+    - Allows print in middle of graph
+    - No dist training
+
+    "dist":
+    - Graph mode by lowering PT program to JAX and then run with Alpa
+    - Doesn't allow print in middle of graph
+    - Supports dist training
+    """
+    assert new_mode in ["local", "dist"]
+    if new_mode == "dist":
+        torch.local_mode = False
```

Comment:
Why do we keep this variable under `torch` instead of `atorch`?

Reply:
It's so that in user module code they can just do `if getattr(torch, "local_mode", True):` and don't need to mention `atorch`, since `atorch` is easy to be confused with `torch` and it's better to mention `atorch` only in trainer code.